### ISSUE TITLE
Too many symbolic links

### ISSUE THREAD

**Issue by bidoubiwa** [eslint/eslint#14151]
**Tell us about your environment**

* **ESLint Version:** 7.21
* **Node Version:** 12
* **npm Version:** 6.14
* **Operating System:** Linux 

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    commonjs: true,
    es2020: true,
    node: true
  },
  globals: {
    strapi: true
  },
  extends: [
    'plugin:react/recommended',
    'standard'
  ],
  parserOptions: {
    ecmaFeatures: {
      jsx: true
    },
    ecmaVersion: 11
  },
  plugins: [
    'react'
  ],
  settings: {
    react: {
      version: 'detect'
    }
  },
  rules: {
    'react/jsx-indent': 'error'
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
eslint --ext .js .
```

```bash
eslint --ext .js .
```

**What did you expect to happen?**
Nothing changed with minor release

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
Oops! Something went wrong! :(

ESLint: 7.21.0

Error: ELOOP: too many symbolic links encountered, 
```

**Steps to reproduce this issue:**

1.  at root, create dir with project
2. Inside sub-dir inside folder, create symbolic link to root (possible recursion?)
3. Run linter and get above throw

**Are you willing to submit a pull request to fix this bug?**
Not sure where the problem lies

**Comment by mdjermanovic** [comment]
Hi @bidoubiwa, thanks for the issue!

The change in behavior is caused by #14126, which fixed #13551 and #13615.

> Nothing changed with minor release

Ignoring symbolic links was a bug introduced in ESLint v7, so we decided to restore the intended behavior in a minor release.

Previous versions used to follow symbolic links, and I'm getting the same `Error: ELOOP: too many symbolic links encountered` with ESLint v6.8.0 if there are circular symlinks.

If you need to have circular links in your project tree, can you ignore them with the [ignorePatterns](https://eslint.org/docs/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files) config option or in the [.eslintignore](https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file) file?

**Comment by bidoubiwa** [comment]
This fixed my issue thanks a lot :)
